libs/ui/popover/src/lib/trigger/popover-trigger.directive.ts

```diff
@@ -200,15 +210,36 @@ export class TsPopoverTriggerDirective implements OnInit, OnDestroy, OnChanges,
     this.popover.referenceObject = this.viewContainerRef.element.nativeElement;
     this.setContentProperties(this.popover);
 
-    // eslint-disable-next-line deprecation/deprecation
-    (fromEvent<MouseEvent>(this.elementRef.nativeElement, 'click'))
-      .pipe(
-        untilComponentDestroyed(this),
-      ).subscribe(event => {
-        this.toggle();
-        event.stopPropagation();
-        event.preventDefault();
-      });
+    switch (this.popoverTrigger) {
```

Comment:
huh the old comments seem to have vanished.. This is definitely style vs functional imo but the general consensus in Angular seems to be that we should keep logic out of `ngOnInit` when possible (just like we would with the constructor) -of course there are exceptions where it's needed for some reason.

In this case I also think it's probably a safer approach to add it to the setter since we have no guarantee that the consumer's input will be set prior to the component being initialized. So if they're waiting on some async data (since everything comes from the store/localstorage/server) the setting value could arrive after initialization.

If, for instance, the logic here also required a second `@Input` to be set to work, then the logic should probably be called from `ngOnChanges`

Reply:
Makes sense! My main concern was that this event subscription shouldn't be called more than once, because otherwise it'd get multiple subscriptions from mouse event and messes it up (that's why it definitely cannot be called from `ngOnChanges`). So as long as consumers aren't setting the trigger multiple times, it's all good.

Reply:
yeah I think either way you'll have to 

a) check for existing sub
b) kill it if it exists
c) create new sub

iirc we have to do the same type of thing somewhere else... maybe input or cohort date?